solutions/simple/main.tf

```diff
@@ -0,0 +1,128 @@
+locals {
+  prefix = var.prefix != null ? (var.prefix != "" ? var.prefix : null) : null
+}
+
+##############################################################################
+# Resource Group
+##############################################################################
+
+module "resource_group" {
+  source                       = "terraform-ibm-modules/resource-group/ibm"
+  version                      = "1.1.6"
+  resource_group_name          = var.use_existing_resource_group == false ? try("${local.prefix}-${var.resource_group_name}", var.resource_group_name) : null
+  existing_resource_group_name = var.use_existing_resource_group == true ? var.resource_group_name : null
+}
+
+#############################################################################
+# COS Bucket for VPC flow logs
+#############################################################################
+
+# parse COS details from the existing COS instance CRN
+module "existing_cos_crn_parser" {
+  count   = var.existing_cos_instance_crn != null ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.existing_cos_instance_crn
+}
+
+locals {
+  bucket_name = try("${local.prefix}-${var.cos_bucket_name}", var.cos_bucket_name)
+
+  bucket_config = [{
+    access_tags                   = var.access_tags
+    bucket_name                   = local.bucket_name
+    kms_encryption_enabled        = var.kms_encryption_enabled_bucket
+    kms_guid                      = var.kms_encryption_enabled_bucket ? module.existing_kms_crn_parser[0].service_instance : null
+    kms_key_crn                   = var.kms_encryption_enabled_bucket ? var.existing_kms_instance_crn : null
+    skip_iam_authorization_policy = var.skip_cos_kms_auth_policy
+    management_endpoint_type      = var.management_endpoint_type_for_bucket
+    storage_class                 = var.cos_bucket_class
+    resource_instance_id          = var.existing_cos_instance_crn
+    region_location               = var.region
+    force_delete                  = true
+  }]
+}
+
+module "cos_buckets" {
+  count          = var.enable_vpc_flow_logs ? 1 : 0
```

Comment:
are we supporting user bringing there own bucket?

Reply:
No, in simple variation...I planned to skip supporting existing storage bucket. So user will bring the COS instance and a new bucket will always be created.

We can discuss...currently it is implemented like this.